boards.txt

```diff
@@ -1321,6 +1321,70 @@ adafruit_hallowing_m4.menu.debug.off=Off
 adafruit_hallowing_m4.menu.debug.on=On
 adafruit_hallowing_m4.menu.debug.on.build.flags.debug=-g
 
+# Adafruit Matrix Portal M4 (SAMD51)
+# ------------------------------
+adafruit_matrixportal_m4.name=Adafruit Matrix Portal M4 (SAMD51)
+adafruit_matrixportal_m4.vid.0=0x239A
+adafruit_matrixportal_m4.pid.0=0x80C9
+adafruit_matrixportal_m4.vid.1=0x239A
+adafruit_matrixportal_m4.pid.1=0x00C9
+adafruit_matrixportal_m4.vid.1=0x239A
+adafruit_matrixportal_m4.pid.1=0x80CA
+adafruit_matrixportal_m4.upload.tool=bossac18
+adafruit_matrixportal_m4.upload.protocol=sam-ba
+adafruit_matrixportal_m4.upload.maximum_size=507904
+adafruit_matrixportal_m4.upload.offset=0x4000
+adafruit_matrixportal_m4.upload.use_1200bps_touch=true
+adafruit_matrixportal_m4.upload.wait_for_upload_port=true
+adafruit_matrixportal_m4.upload.native_usb=true
+adafruit_matrixportal_m4.build.mcu=cortex-m4
+#adafruit_matrixportal_m4.build.f_cpu=120000000L
+adafruit_matrixportal_m4.build.usb_product="Adafruit Matrix Portal M4"
+adafruit_matrixportal_m4.build.usb_manufacturer="Adafruit LLC"
+adafruit_matrixportal_m4.build.board=MATRIXPORTAL_M4
+adafruit_matrixportal_m4.build.core=arduino
+adafruit_matrixportal_m4.build.extra_flags=-D__SAMD51J19A__ -DADAFRUIT_MATRIXPORTAL_M4_EXPRESS -DCRYSTALLESS -D__SAMD51__ {build.usb_flags} -D__FPU_PRESENT -DARM_MATH_CM4 -mfloat-abi=hard -mfpu=fpv4-sp-d16
+adafruit_matrixportal_m4.build.ldscript=linker_scripts/gcc/flash_with_bootloader.ld
+adafruit_matrixportal_m4.build.openocdscript=openocd_scripts/arduino_zero.cfg
+adafruit_matrixportal_m4.build.variant=matrixportal_m4
+adafruit_matrixportal_m4.build.variant_system_lib=
+adafruit_matrixportal_m4.build.vid=0x239A
+adafruit_matrixportal_m4.build.pid=0x80C9
+adafruit_matrixportal_m4.bootloader.tool=openocd
+adafruit_matrixportal_m4.bootloader.file=featherM4/bootloader-feather_m4-v2.0.0-adafruit.5.bin
```

Comment:
It is still draft PR, This is for reminder: it should have its own bootloader.

Reply:
fixed :)